frepple/controllers/outbound.py

```diff
@@ -712,27 +712,25 @@ def export_boms(self):
             "routing_id",
             "type",
             "bom_line_ids",
+            "picking_type_id"
         ]
         for i in bom_recs.read(bom_fields):
             if not i['routing_id']:
                 i['routing_id'] = dummy_mrp_route_m2o_read
 
             # Determine the location
             # We actually want the stock location of the warehouse the location belongs to.
-            if i["routing_id"]:
-                location_id = mrp_routings.get(i["routing_id"][0], None)
-                if not location_id:
-                    location_name = self.env['res.company'].search([
-                        ('name', '=', self.company),
-                    ], limit=1).manufacturing_warehouse.lot_stock_id.complete_name or self.company
-                    # location = self.mfg_location  # Original frePPLe code.
-                else:
-                    location_name = self.env['stock.location'].browse(location_id).get_warehouse_stock_location()
-            else:
-                location_name = self.env['res.company'].search([
-                    ('name', '=', self.company),
-                ], limit=1).manufacturing_warehouse.lot_stock_id.complete_name or self.company
-                # location = self.mfg_location  # Original frePPLe code.
+            location_name = self.env['res.company'].search(
+                [('name', '=', self.company)],
+                limit=1).manufacturing_warehouse.lot_stock_id.complete_name or self.company
+            if i["picking_type_id"]:
+                picking_type = self.env['stock.picking.type'].browse(i["picking_type_id"][0])
+                if picking_type:
+                    location_name = picking_type.warehouse_id.lot_stock_id.complete_name
+            elif i["routing_id"]:
+                    location_id = mrp_routings.get(i["routing_id"][0], None)
+                    if location_id:
+                        location_name = self.env['stock.location'].browse(location_id).get_warehouse_stock_location()
```

Comment:
I've changed the order so that the default value comes last in the `else`, otherwise it was not obvious:

```suggestion
            if i["picking_type_id"]:
                picking_type = self.env['stock.picking.type'].browse(i["picking_type_id"][0])
                if picking_type:
                    location_name = picking_type.warehouse_id.lot_stock_id.complete_name
            elif i["routing_id"]:
                    location_id = mrp_routings.get(i["routing_id"][0], None)
                    if location_id:
                        location_name = self.env['stock.location'].browse(location_id).get_warehouse_stock_location()
            else:
                location_name = self.env['res.company'].search(
                    [('name', '=', self.company)],
                    limit=1).manufacturing_warehouse.lot_stock_id.complete_name or self.company
```

Reply:
@BT-rmartin I've changed the title and description of the PR so that it is readable. Now the description is the old title (formatted) and the title of the PR is self-explanatory without having to open it. In general, I try to [style them](https://cbea.ms/git-commit/).

Reply:
@BT-cserra I won''t do that change as it's wrong. It needs a default value, and if the if and elif, if satisfying the inner if it changes. With your proposal, if the inner if fails, it gets no default value.
Thanks for the reformat in the PR title

Reply:
Well spotted.